docs/psf_spec/high_level_overview.rst

```diff
@@ -0,0 +1,244 @@
+to more detailed descriptions of each block is also provided.
+
+.. graphviz::
+
+    digraph PSFFitting {
+        compound = true;
+        newrank = true;
+        subgraph cluster0 {
+            label="IterativelySubtractedPSFPhotometry"
+            f [label = "finder"];
+            g [label = "group maker"];
+            som [label = "single object model"];
+            n [label = "noise"];
+            ft [label = "fitter"];
+            b [label = "background estimator"];
+            c [label = "culler and ender"];
+            // required to allow finder and guess to combine into group maker
+            invis [shape = circle, width = .1, fixedsize = true, style = invis];
+        }
+
+        w [label = "wcs"];
+        i [label = "input image", shape = box];
+        gs [label = "guess at stars", shape = box];
+        p [label = "psf model"];
+        sm [label = "scene maker"];
+        st [label = "stars\n(astropy.table)", shape = box];
+
+        label = <<FONT POINT-SIZE = "20"><I>photutils.psf </I>class structures</FONT>>;
+        labelloc = top;
+
+        w -> i [style = dashed];
+        // lhead sets the arrow end at the subgraph, rather than at finder itself
+        i -> f [lhead = cluster0];
+        gs -> invis [style = dashed];
+
+        f -> g;
+        f -> invis;
+        invis -> g;
+        som -> f [style = dashed];
+        n -> ft;
+        som -> ft;
+        g -> ft;
+        b -> f [style = dashed];
+        b -> g [style = dashed];
+        b -> ft [style = dashed];
+        ft -> c [style = dashed, label = "Subtract", dir = none];
+        c -> f [style = dashed];
+
+        i -> gs [style = invis];
+        
+        p -> som;
+        sm -> g [style = dotted];
+        ft -> st;
+        // reverse arrow to put psf model above scene maker, saving space with scene
+        // maker next to the middle of the main box
+        p -> sm [style = dotted, dir = back];
+
+        // puts things that should be at the top of the box at the top for orientation
+        // and structuring; requires 'newrank' above to set subgraph items equal
+        {rank = same; w; som;}
+        {rank = same; p; f; n;}
+        {rank = same; gs; g; sm;}
+    }
+
+
+.. _Image:
+
+Input Image
+^^^^^^^^^^^
+
+The input image is the most fundamental part of the PSF fitting process, as it is the
+product on which all subsequent methodology is applied. The image must be a two-dimensional
+array of ``x`` and ``y`` coordinates with ``flux`` represented by array values. If a
+WCS -- contained within a FITS file, for instance -- is passed with the two-dimensional
+array then the position of sources can be given in sky coordinates, instead of pixel values.
+The given image is passed as an input to the chosen PSF fitting class.
+
+
+.. _Finder:
+
+Finder
+^^^^^^
+
+The finder is the first step in the PSF fitting process, as sources must be discovered in
```

Comment:
These all need to be implementation of the `StarFinderBase` class, right? (i.e., https://photutils.readthedocs.io/en/stable/api/photutils.StarFinderBase.html#photutils.StarFinderBase).  If so you should say that, and link to it (e.g. ``  `~photutils.detectionStarFinderBase`  ``)

Reply:
More broadly: I wonder if the `StarFinderBase` should actually be `ObjectFinderBase` or something like that?  Not necessary for this document, but worth thinking a bit about.

Reply:
I've updated the above references of ``stars`` to ``sources`` as that doesn't particularly affect anything and can be future-proofed now, but I obviously can't change these names as they're hard-coded links for now. Where would be best to raise this as a separate issue? Is actually making an issue the right approach for this?